### ISSUE TITLE
Explore other kzg backends

### ISSUE THREAD

**Issue by pawanjay176** [sigp/lighthouse#5025]
## Description

We have newer, and potentially more performant kzg libraries now in
https://github.com/sifraitech/rust-kzg/ and https://github.com/mratsim/constantine/tree/master/constantine-rust/constantine-ethereum-kzg . 

On the consensus layer side, we only use the verification functions on which the benchmarks for the newer libraries are pretty similar to c-kzg. However, using ckzg's verification functions in lighthouse's blob verification functions might be slower than raw benchmarks due to us having to clone the blob before passing it to ckzg.

We need to look into how well different kzg backends perform with lighthouse by gathering metrics on test networks.
Note: Even if the other libs are significantly more performant (which I'm not convinced they would be), we need to consider the fact that other libraries haven't gone through security audits.

**Comment by sauliusgrigaitis** [comment]
Non-batched verification takes only a few milliseconds so obviously there isn't much space for improvement. Batched verification is much faster in `rust-kzg` because of parallelisation, but in a real world setting it's unlikely to have a huge impact on overall client performance.

We use `rust-kzg` in Grandine mainly because it allows switching between different ECC backends as all the supported backends in `rust-kzg` share the same Rust API (C API is the same too, but it's relevant for non-Rust clients). So `rust-kzg` in CL world is more about bringing ECC diversity. Other use cases would benefit from `rust-kzg` performance more.
